File: count_transactions.js
const https = require('https');

function countTransactionsByRoster() {
    const leagueId = "1249067741470539776";
    const url = `https://api.sleeper.app/v1/league/${leagueId}/transactions/1`;
    
    return new Promise((resolve, reject) => {
        https.get(url, (res) => {
            let data = '';
            
            res.on('data', (chunk) => {
                data += chunk;
            });
            
            res.on('end', () => {
                try {
                    const transactions = JSON.parse(data);
                    
                    // Count free agent transactions by roster
                    const rosterCounts = {};
                    let commissisnerCount = 0;
                    
                    transactions.forEach(transaction => {
                        if (transaction.type === 'free_agent') {
                            const rosterId = transaction.roster_ids[0];
                            rosterCounts[rosterId] = (rosterCounts[rosterId] || 0) + 1;
                        } else if (transaction.type === 'commissioner') {
                            commissisnerCount++;
                        }
                    });
                    
                    console.log("Pre-Season Transaction Analysis - 2025 League");
                    console.log("=".repeat(50));
                    console.log("Free Agent Transactions by Roster:");
                    
                    let totalFA = 0;
                    const sortedRosterIds = Object.keys(rosterCounts).sort((a, b) => parseInt(a) - parseInt(b));
                    
                    sortedRosterIds.forEach(rosterId => {
                        const count = rosterCounts[rosterId];
                        totalFA += count;
                        const feesOwed = Math.max(0, count - 10) * 2; // $2 per transaction after 10 free
                        const status = count <= 10 ? "✅ Under limit" : `💰 Owes $${feesOwed}`;
                        console.log(`  Roster ${rosterId}: ${count} transactions - ${status}`);
                    });
                    
                    console.log("\nSummary:");
                    console.log(`  Total Free Agent Transactions: ${totalFA}`);
                    console.log(`  Commissioner Transactions: ${commissisnerCount}`);
                    console.log(`  Total Transactions: ${totalFA + commissisnerCount}`);
                    
                    // Calculate total fees owed
                    const totalFees = Object.values(rosterCounts).reduce((sum, count) => 
                        sum + Math.max(0, count - 10) * 2, 0);
                    
                    if (totalFees > 0) {
                        console.log(`\n💰 Total Fees Owed: $${totalFees}`);
                    } else {
                        console.log(`\n✅ No fees owed - all rosters within free transaction limits`);
                    }
                    
                    resolve();
                } catch (error) {
                    reject(error);
                }
            });
        }).on('error', (error) => {
            reject(error);
        });
    });
}

countTransactionsByRoster().catch(console.error);

File: count_transactions.py
#!/usr/bin/env python3
import json
import urllib.request
import sys

def count_transactions_by_roster():
    """Count free agent transactions by roster for 2025 league"""
    league_id = "1249067741470539776"
    url = f"https://api.sleeper.app/v1/league/{league_id}/transactions/1"
    
    try:
        with urllib.request.urlopen(url) as response:
            transactions = json.loads(response.read())
        
        # Count free agent transactions by roster
        roster_counts = {}
        commissioner_count = 0
        
        for transaction in transactions:
            if transaction['type'] == 'free_agent':
                roster_id = transaction['roster_ids'][0]
                roster_counts[roster_id] = roster_counts.get(roster_id, 0) + 1
            elif transaction['type'] == 'commissioner':
                commissioner_count += 1
        
        print("Pre-Season Transaction Analysis - 2025 League")
        print("=" * 50)
        print(f"Free Agent Transactions by Roster:")
        total_fa = 0
        for roster_id in sorted(roster_counts.keys()):
            count = roster_counts[roster_id]
            total_fa += count
            fees_owed = max(0, count - 10) * 2  # $2 per transaction after 10 free
            status = "✅ Under limit" if count <= 10 else f"💰 Owes ${fees_owed}"
            print(f"  Roster {roster_id}: {count} transactions - {status}")
        
        print(f"\nSummary:")
        print(f"  Total Free Agent Transactions: {total_fa}")
        print(f"  Commissioner Transactions: {commissioner_count}")
        print(f"  Total Transactions: {total_fa + commissioner_count}")
        
        # Calculate total fees owed
        total_fees = sum(max(0, count - 10) * 2 for count in roster_counts.values())
        if total_fees > 0:
            print(f"\n💰 Total Fees Owed: ${total_fees}")
        else:
            print(f"\n✅ No fees owed - all rosters within free transaction limits")
            
    except Exception as e:
        print(f"Error: {e}")
        return 1
    
    return 0

if __name__ == "__main__":
    sys.exit(count_transactions_by_roster())
